src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -981,9 +981,7 @@ public function createService(Definition $definition, $id, $tryProxy = true)
         }
 
         if ($callable = $definition->getConfigurator()) {
-            if (is_array($callable)) {
-                $callable[0] = $callable[0] instanceof Reference ? $this->get((string) $callable[0]) : $parameterBag->resolveValue($callable[0]);
-            }
+            $callable = $this->resolveServices($parameterBag->resolveValue($callable));
```

Comment:
Expressions are not supported in configurators in other places. They are only supported in arguments. So you don't introduce consistency but inconsistency (and using an expression for the service instance of the callable does not make sense either btw)